spring-cloud-starter-task-composedtaskrunner/README.adoc

```diff
@@ -0,0 +1,134 @@
+launch CCC.
+
+=== Splits
+Allows a user to execute tasks in parallel.
+For example:
+```
+<AAA || BBB || CCC>
+```
+Will launch AAA, BBB and CCC in parallel.   When launching splits as a part of a
+composed task all elements of the split must finish successfully before the
+next task definition can be launched for example:
+```
+<AAA || BBB || CCC> && DDD && EEE
+```
+In the case above once AAA, BBB and CCC complete sucessfully then DDD and EEE
+will be launched in the sequence enumerated above.  However if one of the task
+definitions fails in the split then DDD and EEE will not fire.  For example if
+BBB fails then AAA and CCC will be marked successful and BBB will be marked a
+failure and DDD and EEE will not be launched.
+
+== Traversing the graph
+Composed Task Runner is built using
+http://docs.spring.io/spring-batch/reference/html/[Spring Batch]
+to execute the directed graph.   As such each node in the graph is a
+http://docs.spring.io/spring-batch/reference/html/domain.html#domainStep[Step].
+As discussed in the overview, each step in the graph will post a request to a
+Spring Cloud Data Flow Server to execute a task definition.  Each Step has a
+ComposedTaskStepExecutionListener associated with it, such that once the step
+has been executed the
+http://docs.spring.io/spring-batch/reference/htmlsingle/#stepExecutionListener[StepExecutionListener]
+monitors the database awaiting the task to complete.  If the task launched by
+the step fails to complete within the time specified by the `max-wait-time`
+property, the
+http://docs.spring.io/spring-batch/trunk/apidocs/org/springframework/batch/core/ExitStatus.html[ExitStatus]
+of the Step will be set to UNKNOWN.  Once task launched by the step completes,
+the StepExecutionListener will set the ExitStatus of that step based on the following rules:
+
+* If the TaskExecution has an ExitMessage that will be used as the ExitStatus
+* If no ExitMessage is present and the ExitCode is set to 0 then the ExitStatus
+for the step will be COMPLETED.
+* If no ExitMessage is present and the ExitCode is set to 1 then the ExitStatus
+for the step will be FAILED.
+
+If the state of any step in the graph is set to  FAILED and is not handled by
+the DSL the Directed Graph execution will terminate.
+
+
+== Usage
+`java -jar composedtaskrunner-task-1.0.0.BUILD-SNAPSHOT.jar --graph=<your graph syntax>`
+
+== Options
+
+// see syntax (soon to be automatically generated) in spring-cloud-stream starters
+The **$$ComposedTaskRunner$$** $$task$$ has the following options:
+
+//tag::configuration-properties[]
+$$composed-task-arguments$$:: $$The arguments to be used for each of the tasks.$$ *($$String$$, default: `$$<none>$$`)*
+$$composed-task-properties$$:: $$The properties to be used for each of the tasks as well as their deployments.$$ *($$String$$, default: `$$<none>$$`)*
+$$data-flow-uri$$:: $$The URI for the dataflow server that will receive task launch requests.$$ *($$URI$$, default: `$$<none>$$`)*
```

Comment:
Created issue for this: https://github.com/spring-cloud-task-app-starters/composed-task-runner/issues/3